model/src/main/kotlin/FileList.kt

```diff
@@ -64,4 +64,12 @@ data class FileList(
             }
         }
     }
+
```

Comment:
IIUC the meat of `#10502` will be to avoid duplicate entries in "file lists" and "copyright / license findings" of scan results. For my taste, this PR is adding a bit too much lines of code for this undertaking.

Also I have reservations about overloading `+` instead of a `merge()` function. I do not see the benefit, but IMO the `+` has a bit more potential to be misread.

Reply:
> For my taste, this PR is adding a bit too much lines of code for this undertaking.

I was thinking that it makes sense to extract the merging into dedicated functions instead of having it all be part of the `Scanner`. But by doing that, I would like to ensure that the code makes sense even if it is potentially used somewhere else in the future.

* Should we not have this generic code and do it in `Scanner` instead?
* If we keep it generic, which part do you think is not necessary?

> Also I have reservations about overloading + instead of a merge() function.

Sure, I wasn't sure about the customs in Kotlin. I'll change it to a `merge` function.

Reply:
> Also I have reservations about overloading `+` instead of a `merge()` function. I do not see the benefit, but IMO the `+` has a bit more potential to be misread.

I guess it's a matter of taste, but generally I believe we're making too little use of operator overloading as a Kotlin feature, as it usually improves readability (when sticking to semantics that are commonly assumed for those operators).

But to avoid @maennchen going ping-pong to please one reviewer or the other, I'll leave this to you guys.

Reply:
I was using the plus operator on most things that I'm merging in those functions. That's why doing down the operator route seemed to fit well.
But I don't care much either way. Just let me know which direction you prefer :)

Reply:
It's vacation time, and the remaining of us have a lot on their places. So no promises, but I'll try to have a look. Would be great if @mnonnenmacher or @MarcelBochtler could have a look as well.